src/blockdata/block.rs

```diff
@@ -155,6 +157,61 @@ impl BlockHeader {
     }
 }
 
+/// MWEB Block header
+#[derive(PartialEq, Eq, Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct MwebBlockHeader {
+    height: i32,
+    output_root: [u8; 32],
+    kernel_root: [u8; 32],
+    leafset_root: [u8; 32],
+    kernel_offset: [u8; 32], // blinding factor
+    stealth_offset: [u8; 32], // blinding factor
+    output_mmr_size: u64,
+    kernel_mmr_size: u64,
+}
+
+impl consensus::Decodable for MwebBlockHeader {
+    #[inline]
+    fn consensus_decode<D: io::Read>(d:D,) -> Result<MwebBlockHeader, consensus::encode::Error>{
+        let mut d = d.take( consensus::encode::MAX_VEC_SIZE as u64);
+        // not sure why, but have to skip this byte which always has value of 130
+        let _skip = u8::consensus_decode(&mut d)?;
```

Comment:
Value of 130?! Where are you getting that from? It should be `1`, which indicates there is an MWEB block to deserialize, or `0` if there's no MWEB block.